### ISSUE TITLE
Instance isn't saved when trying to migrating to 0.2

### ISSUE THREAD

**Issue by jplehmann** [furious-luke/django-address#30]
Hi Luke, I am finally trying to use your 0.2 branch, and have the migration mostly working, except it's not able to save.  (If you remember we emailed a few months ago, and I am migrating into django-address for the first time).  

```
Traceback (most recent call last):
  File "/Users/john/git/myapp/mysite/products/migrations/0058_auto_20160106_1433.py", line 31, in get_addresses
    new_addr = geolocate(old_addr)
  File "/Users/john/.venv/mm/lib/python2.7/site-packages/address/operations.py", line 20, in geolocate
    return to_python(location.raw, None, self.address_model, self.component_model)
  File "/Users/john/.venv/mm/lib/python2.7/site-packages/address/models.py", line 166, in to_python
    return _to_python(value, instance, address_model, component_model)
  File "/Users/john/.venv/mm/lib/python2.7/site-packages/address/models.py", line 85, in _to_python
    obj.parent = parent
  File "/Users/john/.venv/mm/lib/python2.7/site-packages/django/db/models/fields/related.py", line 668, in __set__
    (value, self.field.rel.to._meta.object_name)
ValueError: Cannot assign "<Component: Component object>": "Component" instance isn't saved in the database.
```

This kind of error usually suggests that we're trying to assign a reference to a foreign key for an object not yet saved.  I don't think it matters but FYI this model has a custom CharField primary key.

My code, adapted from your example is:
```
def get_addresses(apps, schema_editor, geolocate):
    """Load addresses from your database, converting each to
    an address string that can be used in a geolocation lookup.
    """

    my_model = apps.get_model('products.Store')
    for inst in my_model.objects.all():

        if inst.city == None or inst.state == None:
            log.info("Missing city or state for: %s", inst.id)
            continue

        # Get the string of the old address.
        old_addr = u"{}, {} {}, {}".format(
                inst.city, inst.state, inst.zipcode, inst.country.alpha3)

        # Perform a lookup to convert the string into a
        # django-address Address object.
        try:
            new_addr = geolocate(old_addr)
        except:
            log.error("Problem converting store: %s", inst.id, exc_info=True)
            continue

        log.info("Successfully converted: %s", inst.id)

        # Now you can store the address in your system.
        inst.new_address = new_addr
```

**Comment by furious-luke** [comment]
Hi John,

Glad to hear you're starting to look at migrating to 0.2!

So, I'm not sure when you checked out the code, but would you mind double checking that you're using the latest commits to the `dev` branch? I've been making some tweaks lately that may have corrected this issue.

If you still find it's not working I'll set up a test dataset and check that it works from here.

Cheers,
Luke

**Comment by jplehmann** [comment]
Hi Luke, my requirements file is pointed at the latest commit in dev as of
this morning:

+git+
https://github.com/furious-luke/django-address@c30e4b4f9a42897c7da1af488e4dfef3d2a5a894

**Comment by furious-luke** [comment]
Right you are. I'll set up some test data and see if I can replicate the problem.

**Comment by furious-luke** [comment]
Hi again John,

After some digging around (and fixing some other problems I've just noticed), it seems I am unable to replicate your issue. I don't seem to be facing any issues assigning a `Component` to another `Component` when neither of them are saved.

What version of Django are you running? And which DBRM are you using?

Cheers,
Luke

**Comment by furious-luke** [comment]
It looks like Django 1.8.3 added in a check to throw an error if the foreign-key is unsaved. I'm using Django 1.9 and that error no longer seems to be present.

**Comment by jplehmann** [comment]
Am using Django 1.8.2 with Postgresql 9.2.

**Comment by furious-luke** [comment]
Yep, no problems. I know how to disable that Django exception manually, I'll update the code first thing tomorrow and run some tests.

**Comment by jplehmann** [comment]
Trying to upgrade to 1.9 but it may be some work for me due to other libraries' issues. 

I unmigrated, then got your latest change, then migrated again and still have have the same issue, not that you were advertising those changes would fix anything.

**Comment by furious-luke** [comment]
Okay, I've pushed some changes which I *hope* will fix the issue. I've not actually tested on Django 1.8.2, but I think it should do the job. If not I'll stop being lazy and actually install 1.8.2 and test it myself. :)

**Comment by jplehmann** [comment]
Great success, fixed in 5bc3339e62164219bb090b6d7e32b2ed64fbca47.
